src/app/static/src/js/components/runReport/instances.vue

```diff
@@ -64,13 +69,14 @@
             selectInitialInstance: function () {
                 if (this.instances && this.showInstances) {
                     const instances = this.instances;
+                    const initialInstances = this.initialSelectedInstances || {};
                     for (const key in instances) {
                         if (instances[key].length > 0) {
-                            this.$set(this.selectedInstances, key, instances[key][0]);
+                            this.$set(this.selectedInstances, key, initialInstances[key] || instances[key][0]);
                         }
                     }
+                    this.$emit("selectedValues", this.selectedInstances);
```

Comment:
Just to note that the logic here differs slightly from `changeLog.vue`, where we don't emit an event if initial value is used to populate selected value. I can see why we do here - because `selectedValues` could in theory contain a combination of initial and non-initial values. So I guess you could change the `changeLog` code to emit an event, even if it's unlikely that users of the component will be expecting or need such an event.

Reply:
Yes, it is a little inconsistent. Maybe it would be better just to emit here if any new selected values were actually chosen during initialisation?